### PR TITLE
Add Spyre Operation Support Test for Online Softmax

### PR DESCRIPTION
**Pull request by jvlunteren**
## Description

This PR adds a test script to validate Spyre device support for operations required by the online softmax algorithm used in FlashAttention-style fused tiled attention. The script automatically detects unsupported operations (e.g., `torch.maximum`) and implements CPU fallback, enabling full algorithm validation even when certain operations are not yet available on Spyre hardware.

## Related Issues

#121, #147 

## Test Plan

The test plan is integrated into the standalone script, which automatically executes a validation suite including: (1) operation support detection at startup, (2) CPU validation against reference implementation, (3) Spyre execution with hybrid mode, and (4) numerical accuracy verification. Simply run the script to get a report on Spyre's operation support status and algorithm correctness.

## Checklist

- [X] I have read the [contributing guidelines](https://torch-spyre.github.io/spyre-inference/contributing/)
- [X] My code follows the project's code style (run `bash format.sh`)
- [X] I have added tests for my changes (if applicable)
- [X] I have updated the documentation (if applicable)
- [X] My commits include a `Signed-off-by:` line (DCO compliance)

### CODE REVIEW

**Comment by joerunde** [comment]
Nice!

I notice on my machine that this still prints out a lot of warnings about the `aten.argmax.default` falling back to cpu- but I assume we don't care about that because we're only interested in the fact that torch-spyre supports this op, even if it's not run on the cards?

Also there's one warning about a downcast from int64 to int32 during the check for `.max` - I'm not exactly sure what that's referring to though, do you have any idea?

my output:
```
============================================================
Online Softmax — Spyre Operation Support Test
============================================================

Detecting Spyre operation support...
  torch.maximum() — Supported
  torch.bmm() — Supported
  torch.exp() — Supported
/home/senuser/spyre-inference/examples/experimental/spyre_online_softmax_check.py:132: FallbackWarning: aten.argmax.default is falling back to cpu
  a.max(dim=-1, keepdim=True)[0].cpu()
/home/senuser/spyre-inference/examples/experimental/spyre_online_softmax_check.py:132: UserWarning: Backend Spyre does not support int64; downcasting to int32 may change values outside the 32-bit range. You can silence this via warnings.filterwarnings(...) or spyre.set_downcast_warning(False) or TORCH_SPYRE_DOWNCAST_WARN env. variable. (Triggered internally at /home/senuser/.cache/uv/git-v0/checkouts/b06721edd05a1596/c124a56/torch_spyre/csrc/types_mapping.h:96.)
  a.max(dim=-1, keepdim=True)[0].cpu()
  torch.max(dim=-1) — Supported
  torch.sum(dim=-1) — Supported
  arithmetic (mul, add, sub, div) — Supported

Mode: Native Spyre only
============================================================

Creating test inputs:
  Query: [8, 64, 128]
  Key:   [8, 256, 128]
  Value: [8, 256, 128]

Computing reference attention (standard unfused, CPU)...
Computing fused tiled attention (Q_tile=32, KV_tile=64, CPU)...

CPU Validation:
  Max difference:  3.58e-07
  Mean difference: 2.41e-08
  Matches reference!

============================================================
Testing on Spyre Device
============================================================

Creating Spyre tensor layouts and transferring inputs...
  Query: torch.Size([8, 64, 128]), device=spyre:0
  Key:   torch.Size([8, 256, 128]), device=spyre:0
  Value: torch.Size([8, 256, 128]), device=spyre:0

Computing fused tiled attention on Spyre (native Spyre for torch.maximum)...
/home/senuser/spyre-inference/examples/experimental/spyre_online_softmax_check.py:283: FallbackWarning: aten.argmax.default is falling back to cpu
  scores_max = scores.max(dim=-1, keepdim=True)[0]
/home/senuser/spyre-inference/examples/experimental/spyre_online_softmax_check.py:283: FallbackWarning: aten.argmax.default is falling back to cpu
  scores_max = scores.max(dim=-1, keepdim=True)[0]
/home/senuser/spyre-inference/examples/experimental/spyre_online_softmax_check.py:283: FallbackWarning: aten.argmax.default is falling back to cpu
  scores_max = scores.max(dim=-1, keepdim=True)[0]
/home/senuser/spyre-inference/examples/experimental/spyre_online_softmax_check.py:283: FallbackWarning: aten.argmax.default is falling back to cpu
  scores_max = scores.max(dim=-1, keepdim=True)[0]

Spyre Validation:
  Max difference:  3.70e-03
  Mean difference: 2.69e-04
  Matches reference within fp16 tolerance (1e-02)

============================================================
Test Complete
============================================================
Spyre operation support summary:
  Supported: bmm, mul, add, sub, div, exp, sum(dim=-1), max(dim=-1)
  Not supported: (none detected)

  Known bugs requiring workarounds:
    - seq-dim narrowed-view reads on Spyre (bmm yields garbage)
      workaround: slice on host then transfer (_seq_slice)
    - seq-dim narrowed-view writes on Spyre (data corruption)
      workaround: write via CPU round-trip (fused_tiled_attention line 340-344)
```

**Comment by jvlunteren** [comment]
Both warnings are expected and harmless.

- The `aten.argmax.default` CPU fallback: `torch.max(dim=-1)` decomposes into `amax` (Spyre-native, the values) + `argmax` (CPU fallback, the indices). We only use the values,  so the indices fallback is dead work. But yes, we only care that the values path runs on Spyre, which it does.

- The int64 downcast warning comes from the `argmax` return value landing on device. Spyre doesn't support int64 so it silently narrows to int32. Same root cause, same conclusion: indices path we don't use, irrelevant to the test.

**Review by joerunde — APPROVED** [review]
lgtm then!
